src/EditorFeatures/VisualBasic/LineCommit/CommitFormatter.vb

```diff
@@ -87,6 +90,20 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
             End Using
         End Sub
 
+        Private Function AbortForDiagnostics(document As Document, textSpanToFormat As TextSpan, cancellationToken As CancellationToken) As Boolean
+            Const UnterminatedStringId = "BC30648"
+
+            Dim tree = document.GetSyntaxTreeAsync(cancellationToken).WaitAndGetResult(cancellationToken)
+
+            ' If we have any unterminated strings that overlap what we're trying to format, then
+            ' bail out.  It's quite likely the unterminated string will cause a bunch of code to
```

Comment:
Rather than bailing the entire format, should we still format the stuff up to (but not including) where the unterminated string literal starts? Consider I copy in some large snippet, and miss the closing ". We might still want to format other stuff. Also, consider the user experience once I do fix the ". Do you want to remember all the other stuff was dirty and it should finally format?

I'm asking only because the alternative approaches would mean the span calculation here should move to `CommitBufferManager.TryComputeExpandedSpanToFormat`, or perhaps there should be some bail in `CommitBufferManager.CommitDirty`. I bring this up mostly because whenever somebody touches this place or one of those functions, I always ask myself whether it should really be somewhere else. And honestly in this case I don't have a strong sense what the right user experience is here.